rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -204,42 +209,42 @@ rmw_create_client(
   return rmw_client;
 
 fail:
+  if (info->request_publisher_ != nullptr) {
+    Domain::removePublisher(info->request_publisher_);
+  }
 
-  if (info != nullptr) {
-    if (info->request_publisher_ != nullptr) {
-      Domain::removePublisher(info->request_publisher_);
-    }
+  if (info->response_subscriber_ != nullptr) {
+    Domain::removeSubscriber(info->response_subscriber_);
+  }
 
-    if (info->response_subscriber_ != nullptr) {
-      Domain::removeSubscriber(info->response_subscriber_);
-    }
+  if (info->listener_ != nullptr) {
+    delete info->listener_;
+  }
 
-    if (info->listener_ != nullptr) {
-      delete info->listener_;
+  if (impl) {
+    if (info->request_type_support_ != nullptr) {
+      _unregister_type(participant, info->request_type_support_, info->typesupport_identifier_);
     }
 
-    if (impl) {
-      if (info->request_type_support_ != nullptr) {
-        _unregister_type(participant, info->request_type_support_, info->typesupport_identifier_);
-      }
-
-      if (info->response_type_support_ != nullptr) {
-        _unregister_type(participant, info->response_type_support_, info->typesupport_identifier_);
-      }
-    } else {
-      RCUTILS_LOG_ERROR_NAMED(
-        "rmw_fastrtps_cpp",
-        "leaking type support objects because node impl is null")
+    if (info->response_type_support_ != nullptr) {
+      _unregister_type(participant, info->response_type_support_, info->typesupport_identifier_);
     }
-
-    delete info;
+  } else {
+    RCUTILS_LOG_ERROR_NAMED(
+      "rmw_fastrtps_cpp",
+      "leaking type support objects because node impl is null")
   }
 
-  if (rmw_client->service_name != nullptr) {
-    rmw_free(const_cast<char *>(rmw_client->service_name));
-    rmw_client->service_name = nullptr;
+  delete info;
+  info = nullptr;
+
+  if (nullptr != client) {
```

Comment:
`client`? Is this supposed to be `rmw_client`?

Did you build the code locally? If not, please do so in the future as well as run the tests for the changed code locally.

Reply:
please ignore the modification from rmw_client.cpp and it's added here exceptionally, because it has nothing to do with this topic and it should be in another PR to avoid misleading. thanks.